exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/entities/EntityTests.kt

```diff
@@ -844,6 +844,9 @@ class EntityTests : DatabaseTestsBase() {
                 name = "Eton"
                 region = region1
                 secondaryRegion = region2
+            }.apply {
+                // otherwise Oracle provides school1.id = 0 to testCache(), which returns null
+                if (currentDialectTest is OracleDialect) flush()
```

Comment:
1. What does `flush` mean? Maybe I missed it, but I couldn't find anything about it in the documentation.
2. Do you think this test reflects real-life usage? If we expect users to use Exposed like this with Oracle, what do you think of adding this to the documentation?

Reply:
When `new()` is used, it creates an entity but then just queues it in a cache for insertion (even though SQL statements are logged). `flush()` manually takes the entities in the cache (as well as any entities that have been queued for update) and sends the values to the database. This happens under-the-hood at the end of a transaction or before/during certain other functions that force access to the entities.

I don't think this is a real-life scenario for any users, using `testCache()`, as anything that would access the created entity would trigger a flush. For example, if the `apply {}` block was removed and I did something like `findById()` or even just a `println(school1.id)`, the test would pass.